modules/ximgproc/test/test_bilateral_texture_filter.cpp

```diff
@@ -0,0 +1,141 @@
+ *  the use of this software, even if advised of the possibility of such damage.
+ */
+
+#include "test_precomp.hpp"
+
+namespace cvtest
+{
+
+using namespace std;
+using namespace std::tr1;
+using namespace testing;
+using namespace perf;
+using namespace cv;
+using namespace cv::ximgproc;
+
+//////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////
+
+typedef tuple<int, double, double, MatType, int> BTFParams;
+typedef TestWithParam<BTFParams> BilateralTextureFilterTest;
+
+TEST_P(BilateralTextureFilterTest, SplatSurfaceAccuracy)
+{
+  BTFParams params = GetParam();
+  int fr            = get<0>(params);
+  double sigmaAlpha = get<1>(params);
+  double sigmaAvg   = get<2>(params);
+  int depth         = get<3>(params);
+  int srcCn         = get<4>(params);
+
+  RNG rnd(0);
+
+  Size sz(rnd.uniform(256,512), rnd.uniform(256,512));
+
+  for (int i = 0; i < 5; i++)
+  {
+    Scalar surfaceValue;
+    if(depth == CV_8U)
+        rnd.fill(surfaceValue, RNG::UNIFORM, 0, 255);
+    else
+        rnd.fill(surfaceValue, RNG::UNIFORM, 0.0f, 1.0f);
+
+    Mat src(sz, CV_MAKE_TYPE(depth, srcCn), surfaceValue);
+
+    Mat res;
+    bilateralTextureFilter(src, res, fr, 1, sigmaAlpha, sigmaAvg);
+
+    double normL1 = cvtest::norm(src, res, NORM_L1)/src.total()/src.channels();
+    EXPECT_LE(normL1, 1.0/64.0);
+  }
+}
+
+TEST_P(BilateralTextureFilterTest, MultiThreadReproducibility)
+{
+    if (cv::getNumberOfCPUs() == 1)
+      return;
+
+    BTFParams params = GetParam();
+    int fr            = get<0>(params);
+    double sigmaAlpha = get<1>(params);
+    double sigmaAvg   = get<2>(params);
+    int depth         = get<3>(params);
+    int srcCn         = get<4>(params);
+
+    double MAX_DIF = 1.0;
+    double MAX_MEAN_DIF = 1.0 / 64.0;
+    int loopsCount = 2;
+    RNG rnd(1);
+
+    Size sz(rnd.uniform(256,512), rnd.uniform(256,512));
+
+    Mat src(sz,CV_MAKE_TYPE(depth, srcCn));
+    if(src.depth()==CV_8U)
+        randu(src, 0, 255);
+    else if(src.depth()==CV_16S)
+        randu(src, -32767, 32767);
+    else
+        randu(src, 0.0f, 1.0f);
+
+    for (int iter = 0; iter <= loopsCount; iter++)
+    {
+        cv::setNumThreads(cv::getNumberOfCPUs());
+        Mat resMultiThread;
+        bilateralTextureFilter(src, resMultiThread, fr, 1, sigmaAlpha, sigmaAvg);
+
+        cv::setNumThreads(1);
+        Mat resSingleThread;
+        bilateralTextureFilter(src, resSingleThread, fr, 1, sigmaAlpha, sigmaAvg);
+
+        EXPECT_LE(cv::norm(resSingleThread, resMultiThread, NORM_INF), MAX_DIF);
+        EXPECT_LE(
+          cv::norm(resSingleThread, resMultiThread, NORM_L1),
+          MAX_MEAN_DIF * src.total() * src.channels());
+    }
+}
+
+INSTANTIATE_TEST_CASE_P(
+  TypicalSet1,
+  BilateralTextureFilterTest,
+  Combine(
+    Values(2, 3),
+    Values(0.5, 1.0),
+    Values(0.5, 1.0),
+    Values(CV_8U, CV_32F),
+    Values(1, 3)
```

Comment:
Could you please reduce this test set also (it runs for 2 min - it is about 50% of whole [test_ximgproc](http://pullrequest.opencv.org/buildbot/builders/precommit-contrib_linux64/builds/2293/steps/test_ximgproc))